ee/tuf/autoupdate.go

```diff
@@ -44,10 +46,41 @@ const (
 
 var binaries = []autoupdatableBinary{binaryLauncher, binaryOsqueryd}
```

Comment:
Do you think it's worth an array here, vs a couple of booleans? My gut sense is that we have 2 binaries now, and I can maybe imagine a 3rd. But it's not an unbounded list. So if the code is cleaner with a couple booleans...

Reply:
This whole package is designed to accommodate a list of arbitrary binaries, so it's more consistent and cleaner to keep doing it this way.

Reply:
I guess I'm also avoiding suggesting that we go the protobuf route

Reply:
I meant that `var binaries []autoupdatableBinary` is already used extensively throughout this package, so it would be more consistent for these changes to also use an array rather than a couple of booleans. I didn't realize you were discussing the control <=> launcher data format and not the one used internally in `ee/tuf`.

I think this option is a little more flexible than having the control server send down booleans -- I think if we ended up wanting a "lock to version X.Y.Z for N days" then this data format works well.

Reply:
Oh. Yeah, I'm thinking only about the API layer

Reply:
> lock to version X.Y.Z for N days

How? If we send down an array of things to update, we can't list a separate versions for them. 

Honestly, I'd might keep "check for updates now" and "use version X for N days" separate. K2 can always send two actions.

Reply:
> How? If we send down an array of things to update, we can't list a separate versions for them.

Something like --

```
{
  "binaries_to_update": [
    {
      "name": "launcher",
      "lock_version": "v1.6.0", // omitempty
      "lock_expires": 123456789 // omitempty
    }
  ]
}
```

> Honestly, I'd might keep "check for updates now" and "use version X for N days" separate.

`use version X for N days` requires a check for updates (to confirm if version X is already downloaded and to download it if not), so they fit together nicely.

Reply:
(Which I see is what you did)